test/recipes/25-test_x509.t

```diff
@@ -232,3 +232,21 @@ SKIP: {
         if disabled("ec");
     ok(run(test(["x509_test"])), "running x509_test");
 }
+
+{
+    # Tests for issue #21403 - within isolated variable scope
+    my $i21403_cnf = srctop_file('test', 'test.cnf');
+    my $i21403_key = srctop_file('test', 'key-21403.pem');
+    my $i21403_cert = srctop_file('test', 'cert-21403.pem');
```

Comment:
Updated directory target for key and cert files from 'test' to 'test-runs', and pushed to my repo.

Reply:
@bernd-edlinger

I'm new to the codebase and my code was just a possible fix, but I'm willing to defer to your judgment.

My concern is that this: if sending a particular output to the cert file breaks generation of a type of certificate, doesn't that suggest that it may not be the best place to send that particular output (for this, or for the other options you mentioned)?

Reply:
No, the problem was just that the line-ending was sent to the wrong file
and thus the line containing the modulus was merged with the folllowing
line containing the string `-----BEGIN CERTIFICATE-----`
but that is the key-word that starts the data part of the PEM file,
and therefore the decryption failed.